### DISCUSSION TITLE
How to avoid getting blank pages after plots in pdf output ?

### DISCUSSION THREAD

**Discussion by sunvitD** [quarto-dev/quarto-cli#11721] [Q&A]
### Description

Output in pdf is generating `blank pages` after almost every plot. Its the first time I have created a `pdf` output using quarto and I have been using almost the same code for `html` but html files had no such issues.

1. I am creating multiple charts using multiple tabs (around 10) and 1 chart in each tab. After most of these charts there is a blank page in pdf output and then the next chart again. 1-2  charts do not have any blank pages between them but most of them do.

2. I have another set of tabs that have tables in them and they do not have any such issues so it doesn't look like an issue because of tabs

Sharing the header:

````
---
title: "Analytics Report"
author: "HQ"
format: 
  pdf:
    fontsize: 10pt
    code-overflow: wrap
embed-resources: true
smooth-scroll: true
code-tools:
    source: false
    toggle: false
    caption: none
toc: true
toc-location: right-body
echo: False
warning: False
message: false
jupyter: python3
theme:
  light: lux
  dark: slate
---
````

Unfortunately the `Quarto version` I have is `1.5.57`

Please suggest how to get rid off the empty blank pages from the pdf output. 

Thanks !!

**Comment by mcanouil** [comment]
Unfortunately, if you don't provide a small reproducible example, you'll unlikely get any help.

Note that `embed-resources` and `smooth-scroll` have no meaning except for HTML-based formats. Same goes for `theme`.
Even if Quarto tries to be smart, `echo`, `warning`, `message` should be placed under `execute` as documented (https://quarto.org/docs/computations/execution-options.html).
You showed a snippet for PDF/LaTeX but are talking about "tabs". Only HTML supports tabset.

When using multiple formats, you can set the options under the specific format:

```yaml
format:
  html:
    embed-resources: true
  pdf:
    toc-location: right
 ```


When working with LaTeX, you can (and should) take a look at the LaTeX code to understand what's going on.
You can also take a look at the intermediary markdown to see what your code is actually emitting.

```yaml
keep-md: true
keep-tex: true
```

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by sunvitD** [reply]
Thanks for all the suggestions, I was just testing if the code runs for pdf file output or not & noticed this issue. I will incorporate all these suggestions after solving this issue. 

And I am not sure if this error will be captured in some small demo code and since it contains multiple files `.qmd, .csv, .txt` to reproduce so sharing the  [Github link](https://github.com/sunvitD/Quarto_blankpdf/tree/main). I have also uploaded the output pdf and quarto_check.txt file in there as well.

Sharing the `quarto check` output below as well:

````
Quarto 1.5.57
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.58.3: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.5.57
      Path: C:\Users\vinee\anaconda3\envs\HQ2\Library\bin\
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: v2024.12
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\vinee\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2024

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.12.4 (Conda)
      Path: C:/Users/vinee/anaconda3/envs/HQ2/python.exe
      Jupyter: 5.7.2
      Kernels: python3, pyspark-env

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.3.1
      Path: C:/PROGRA~1/R/R-43~1.1
      LibPaths:
        - C:/Users/vinee/AppData/Local/R/win-library/4.3
        - C:/Program Files/R/R-4.3.1/library
      knitr: (None)
      rmarkdown: (None)

      The knitr package is not available in this R installation.
      Install with install.packages("knitr")
      The rmarkdown package is not available in this R installation.
      Install with install.packages("rmarkdown")
````

Hope this is useful to debug. Please let me know if anything else is required. 

Thanks alot !!

**Reply by mcanouil** [reply]
You should be able to "debug" or at least end up with a small example showing your issue.
I'm afraid we don't have the time to do this standard debugging step for you.
